analysis/estimate/notebooks/compare-estimate-consensus.Rmd

```diff
@@ -377,17 +377,51 @@ make_scatterplot <- function(combined_wide_df, estimate_col, celltype_col, type,
     ) +
   theme(legend.position = "bottom")
 }
+```
+
+
+```{r fig.height=5, fig.width=12}
+combined_wide_df <- combined_wide_df |>
+  dplyr::filter(sample_quality == "acceptable quality") |>
+  dplyr::mutate(
+    seq_unit = ifelse(
+      project_id %in% c("SCPCP000001", "SCPCP000002"), 
+      "single-cell", 
+      "single-nuclei"
+    ), 
+    estimate_score = estimate_score_stromal + estimate_score_immune
+  )
+
+pseudo_wide_df <- combined_wide_df |>
+  dplyr::filter(expression_type == "pseudobulk")
+
+
 patchwork::wrap_plots(
-  make_scatterplot(combined_wide_df, "estimate_score_immune", "celltype_immune", "immune", -1700, 0.75),
-  make_scatterplot(combined_wide_df, "estimate_score_stromal", "celltype_stromal", "stromal", -1800, 0.25),
-  make_scatterplot(combined_wide_df, "estimate_score", "celltype_other", "", -3000, 0.6), 
+  make_scatterplot(pseudo_wide_df, "estimate_score_immune", "celltype_immune", "immune", -1700, 0.75),
+  make_scatterplot(pseudo_wide_df, "estimate_score_stromal", "celltype_stromal", "stromal", -1800, 0.25),
+  make_scatterplot(pseudo_wide_df, "estimate_score", "celltype_other", "", -3000, 0.6), 
   guides = "collect"
-) & theme(legend.position = "bottom")
+) +
+  patchwork::plot_annotation(title = "ESTIMATE from pseudobulk") & theme(legend.position = "bottom")
 ```
 
+```{r fig.height=5, fig.width=12}
+bulk_wide_df <- combined_wide_df |>
+  dplyr::filter(expression_type == "bulk_counts")
+
+
+patchwork::wrap_plots(
+  make_scatterplot(bulk_wide_df, "estimate_score_immune", "celltype_immune", "immune", -1700, 0.75),
+  make_scatterplot(bulk_wide_df, "estimate_score_stromal", "celltype_stromal", "stromal", -1800, 0.25),
+  make_scatterplot(bulk_wide_df, "estimate_score", "celltype_other", "", -3000, 0.6), 
+  guides = "collect"
+) +
+  patchwork::plot_annotation(title = "ESTIMATE from bulk") & theme(legend.position = "bottom")
```

Comment:
```suggestion
expression_types <- c("pseudobulk", "bulk_counts") 
expression_types |> 
purrr::map(\(type){

  filter_df <- combined_wide_df |>
  dplyr::filter(expression_type == type)


patchwork::wrap_plots(
  make_scatterplot(filter_df, "estimate_score_immune", "celltype_immune", "immune", -1700, 0.75),
  make_scatterplot(filter_df, "estimate_score_stromal", "celltype_stromal", "stromal", -1800, 0.25),
  make_scatterplot(filter_df, "estimate_score", "celltype_other", "", -3000, 0.6), 
  guides = "collect"
) +
  patchwork::plot_annotation(title = glue::glue("ESTIMATE from {type}")) & theme(legend.position = "bottom")

}
```
You could simplify this and remove the duplicate code by doing the above. But GitHub spacing is weird so that will need to be adjusted.